### PR TITLE
Fixes #8431: Create a tool to upmerge PR fro github

### PR DESCRIPTION
**Pull request by peckpeck**
https://www.rudder-project.org/redmine/issues/8431

### CODE REVIEW

File: scripts/rudder-dev/github.py [hunk 1]
```diff
@@ -33,6 +49,16 @@ def base_branch(self):
...
 
+  def comment(self, comment):
+    url = "https://api.github.com/repos/Normation/{repo}/issues/{pr_id}/comments"
+    data = { "body": comment }
+    github_request(url, "Posting comment", self.url, json.dumps(data), repo=self.repo)
+
+  def label(self, label):
+    url = "https://api.github.com/repos/Normation/{repo}/issues/{pr_id}/labels"
+    data = [ label ]
+    github_request(url, "Posting comment", self.url, json.dumps(data), repo=self.repo)
```
Comment (VinceMacBuche) [inline scripts/rudder-dev/github.py]: Should it be "Changing Label" instead of "Posting comment" ?

File: scripts/rudder-dev/quality-assistant [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+      # PR must be manually merged
+      comment="""This PR is not mergeable to upper versions.
+Since it is "Ready for merge" you must merge it by yourself using the following command:
+rudder-dev merge """ + pr_info['html_url'] 
+      pr.comment(comment)
+      pr.label(Config.BOT_CANNOT_MERGE)
+    else:
+      # PR can be automatically merged
+      pr=PR(pr_info['html_url'])
+      shell(command + redirect, "Automatically merging PR " + pr_info['html_url'])
```
Comment (VinceMacBuche) [inline scripts/rudder-dev/quality-assistant]: There should be a comment on the pull request stating that is was autmatically merged :)

**Comment by peckpeck** [comment]
Commit modified

**Comment by VinceMacBuche** [comment]
Ready for merge !

**Comment by peckpeck** [comment]
Commit modified

**Comment by peckpeck** [comment]
Commit modified

**Comment by peckpeck** [comment]
Commit modified

**Comment by peckpeck** [comment]
Commit modified

**Comment by peckpeck** [comment]
Commit modified

**Comment by peckpeck** [comment]
This PR is not mergeable to upper versions.
Since it is "Ready for merge" you must merge it by yourself using the following command:
rudder-dev merge https://github.com/Normation/rudder-tools/pull/158
-- Your faithful QA

**Comment by peckpeck** [comment]
I merge and upmerged your PR automatically -- Your faithful QA

**Pull request merged**